src/tensors/cpu/wasm_intgemm_interface.h

```diff
@@ -0,0 +1,246 @@
+ * @param[in]   cols_B                 No. of columns of Input matrix B. Should be a multiple of 8.
+ * @param[out]  output                 An array representing the prepared B matrix.
+ *                                     Size of the array = `width` * `cols_B`.
+ */
+void int8PrepareBFromTransposed(const float* input_B_transposed,
+                                float scale,
+                                float zero_point,
+                                Index width,
+                                Index cols_B,
+                                int8_t* output);
+
+/**
+ * Prepare B for the Matrix Multiply function from a quantized and transposed version of Input
+ * matrix B which is also in a CPU-independent format.
+ *
+ * The final prepared B is in CPU-dependent format and can be used as an input to matrix multiply
+ * function (`int8MultiplyAndAddBias`).
+ *
+ * This function is useful while using the quantized models that are stored in a CPU-independent
+ * format on the disk.
+ *
+ * @param[in]   input_B_quant_transposed   An array representing the quantized and transposed
+ *                                         version of Input matrix B. It is in column-major format.
+ *                                         Size of the array = `width` * `cols_B`.
+ *                                         Shape of the matrix: (`cols_B`, `width`)
+ * @param[in]   width                      No. of rows of Input matrix B. Should be multiple of 64
+ * @param[in]   cols_B                     No. of columns of Input matrix B. Should be multiple of 8
+ * @param[out]  output                     An array representing the prepared B matrix.
+ *                                         Size of the array = `width` * `cols_B`.
+ */
+void int8PrepareBFromQuantizedTransposed(const int8_t* input_B_quant_transposed,
+                                         Index width,
+                                         Index cols_B,
+                                         int8_t* output);
+
+/**
+ * Prepare A for the Matrix Multiply function from Input matrix A.
+ *
+ * It performs quantization on floating values of input.
+ * The final prepared A can be used as an input to matrix multiply function
+ * (`int8MultiplyAndAddBias`).
+ *
+ * Please note that this interface might have architecture specific implementation.
+ *
+ * @param[in]   input_A        An array representing the Input matrix A in row-major format.
+ *                             Size of the array = `rows_A` * `width`.
+ *                             Shape of the matrix: (`rows_A`, `width`)
+ * @param[in]   scale          The scaling factor (for quantization)
+ * @param[in]   zero_point     The zero point (for quantization)
+ * @param[in]   rows_A         No. of rows of Input matrix A. No restriction on its size.
+ * @param[in]   width          No. of columns of Input matrix A. It should be a multiple of 64.
+ * @param[out]  output         An array representing the prepared A matrix.
```

Comment:
Should  mention that on some architectures (x86) the output would be `uint8_t` (Which is achieved by adding 127 to every item in A after quantising), in case someone decides to quantise `A` without using this routine.  

We don't need to change the data type, as we can do ptr casting underneath, but we should probably mention that different architectures will have different underlying behaviour, which will be opaque to the user if they strictly use only the provided functions.

Reply:
> in case someone decides to quantise A without using this routine.

@XapaJIaMnu  I have mentioned in the documentation that the precondition to use the Multiply function is that A, B and Bias should be prepared using Prepare* functions of this interface otherwise there is no guarantee of the correctness of the result. Should I still add the information regarding prepared A to be `uint8` for x86?

Reply:
I believe that additional documentation never hurts. We can have an additional implementation details at the bottom of the document, that may help devs understand better what we are doing. After all our goal is not to produce an obfuscated binary interface. I can write this at a later point, but I think this is an ok interface.

What are the next steps? We need to write implementations of those functions that map to the actual intgemm calls?

Reply:
> We can have an additional implementation details at the bottom of the document, that may help devs understand better what we are doing. After all our goal is not to produce an obfuscated binary interface. I can write this at a later point, but I think this is an ok interface.

okay.




> We need to write implementations of those functions that map to the actual intgemm calls?

Yes, this is the next step.

I am merging the PR then 👍